articles/android-workmanager/index.md

```diff
@@ -0,0 +1,165 @@
+
+```Kotlin
+  // Define OneTime work
+  private val oneTimeWorker = OneTimeWorkRequest.Builder(Work::class.java)
+      .setConstraints(constraints)
+      .build()
+```
+
+**2. Periodic work**
+Sometimes we may need the work to be run at intervals. In this case, we use the `PeriodicWorkRequest` class. This takes in three parameters i.e. our worker class, the interval and the `TimeUnit`. Add the following code.
+
+```Kotlin
+// Define Periodic work
+private val periodicWork = PeriodicWorkRequest.Builder(Work::class.java, 15, TimeUnit.MINUTES)
+    .setConstraints(constraints)
+    .build()
+```
+
+We set our time interval to 15 minutes. This is the minimum interval defined in the [documentation](https://developer.android.com/reference/kotlin/androidx/work/PeriodicWorkRequest).
+
+To start our work, we pass in the defined work into the `WorkManager` instance we created. We call the `enqueue()` method and pass in our work. Since we have multiple jobs defined, we can pass in a list of jobs as a parameter. Create a function called `startWork()` and add the following code.
+
+```Kotlin
+fun startWork(){
+    manager.enqueue(listOf(oneTimeWorker, periodicWork))
+}
+```
+
+This starts the work for us provided the constraints are met.
+
+Sometimes you may want to chain work i.e. start with one job and after completion, begin a second job. Then we make use of `beginWith()` and `then()` methods. Each of these methods receive one job as a parameter. They however receive work of `OneTimeWorkRequest` type only. So periodic work cannot be used in a chain.
+
+### Step 4 - Finishing up
+Now that we have our work completely set up. You can go ahead and follow through the repository to check the UI setup.
+
+Once the application starts, it calls the `startWork` function which starts our work. This starts both the `oneTimeWork` and `periodicWork`. The one time work completes immediately and adds one user to the Room database. The `periodicWork` also starts by adding one user to the database and another user after 15 minutes. So after 2 hours, the `periodicWork` will have added 8 users. On the first run the application should show 2 users. One from the `oneTimeWork` and another from the `periodicWork`.
+
+### Conclusion
+With that, you have the basic information about `WorkManager`. As you can see, it is easier to schedule background tasks. You are able to run tasks like network calls even if the application closes. A good use case could be backing up data on a different network like cloud services. This can be done when some constraints are met and you have the assurance of your work being done. Go ahead and clone the [repo](https://github.com/LinusMuema/kotlin/tree/workManager) and run the application. Feel free to raise a PR or an issue with any updates.
```

Comment:
I'd suggest you let the reader download the repo at the beginning (first step). Then you walk the reader through adding the different classes and methods associated with the technology. I think that would be better.

Reply:
On it...